packages/mui-material/src/Menu/Menu.js

```diff
@@ -164,12 +163,14 @@ const Menu = React.forwardRef(function Menu(inProps, ref) {
         horizontal: isRtl ? 'right' : 'left',
       }}
       transformOrigin={isRtl ? RTL_ORIGIN : LTR_ORIGIN}
-      PaperProps={{
-        as: MenuPaper,
-        ...PaperProps,
-        classes: {
-          ...PaperProps.classes,
-          root: classes.paper,
+      slots={{ paper: MenuPaper }}
+      slotProps={{
+        paper: {
+          ...PaperProps,
+          classes: {
+            ...PaperProps.classes,
+            root: classes.paper,
+          },
         },
       }}
```

Comment:
Hi @DiegoAndai 

I found issue in here. When `SelectInput` component pass `slotProps` to this component. It will make `slotProps` appear in `other` variable at line 80. so we can lose value.

Here is example to give picture when it can happen.
```
<Select
  ...
  MenuProps={{
    slotProps: {
      root: {
        slotProps: {
          backdrop: {
            invisible: true,
          },
        },
      },
    },
  }}
>
  some item...
</Select>
```

I think you should extract `slotProps` and merge at this line.

Reply:
Hi @ethaizone, you're correct!

I overlooked that when adding the `slots` and `slotProps` to the `Popover` component, those would also be made available to the Menu component. If a user uses either, we would lose [the values provided in the Menu component](https://github.com/DiegoAndai/material-ui/blob/master/packages/mui-material/src/Menu/Menu.js#L166-L175).

I created an issue and will be working on it next week: https://github.com/mui/material-ui/issues/37612

Thanks for spotting this 😊